src/Database/Type/JsonType.php

```diff
@@ -50,7 +50,12 @@ public function toDatabase(mixed $value, DriverInterface $driver): ?string
             return null;
         }
 
-        return json_encode($value, $this->_encodingOptions);
+        $value = json_encode($value, $this->_encodingOptions);
+        if ($value === false) {
+            return null;
```

Comment:
I think this should throw an exception since it's invalid input. We don't want to silently encode null, do we?

Reply:
It would be better to simply add `JSON_THROW_ON_ERROR` flag to encoding options instead of doing `false` checks. We do the same in `JsonView`.

Reply:
Sure, but I guess the tooling would still think we do not catch the return false?

Reply:
Yes, but then we can annotate the return type as string and avoid adding unneeded additional code.

Reply:
Turns out PHPStan/Psalm is smart enough to know that with the flag set it cannot return false anymore :)